File: app/src/main/java/com/stupidtree/hita/ui/timetable/views/TimeTableBlockAddView.java
//package com.stupidtree.hita.ui.timetable.views;
//
//import android.view.View;
//import android.view.ViewGroup;
//import android.widget.FrameLayout;
//
//import androidx.annotation.NonNull;
//
//import com.stupidtree.hita.R;
//import com.stupidtree.hita.activities.BaseActivity;
//import com.stupidtree.hita.fragments.popup.FragmentAddEvent;
//import com.stupidtree.hita.timetable.packable.TimePeriod;
//
//public class TimeTableBlockAddView extends FrameLayout {
//    View card;
//    View add;
//    int dow;
//    int week;
//    TimePeriod timePeriod;
//
//    public TimeTableBlockAddView(@NonNull final BaseActivity context, final int week, final int dow, final TimePeriod timePeriod) {
//        super(context);
//        this.dow = dow;
//        this.week = week;
//        this.timePeriod = timePeriod;
//        inflate(context, R.layout.dynamic_timetable_block_add,this);
//        add = findViewById(R.id.add);
//        card = findViewById(R.id.card);
//        add.setOnClickListener(new OnClickListener() {
//            @Override
//            public void onClick(View view) {
//                new FragmentAddEvent().setInitialData(week,dow,timePeriod).show(context.getSupportFragmentManager(),"fae");
//                ViewGroup parent = (ViewGroup) getParent();
//                if(parent!=null) parent.removeView(com.stupidtree.hita.views.TimeTableBlockAddView.this);
//            }
//        });
//        card.setOnClickListener(new OnClickListener() {
//            @Override
//            public void onClick(View view) {
//                add.callOnClick();
//            }
//        });
//    }
//
//    public int getDow() {
//        return dow;
//    }
//
//    public TimePeriod getTimePeriod() {
//        return timePeriod;
//    }
//
//    public int getDuration(){
//        return timePeriod==null?0:timePeriod.getLength();
//    }
//}

File: app/src/main/java/com/stupidtree/hita/ui/main/timetable/views/TimeTableBlockAddView.kt
package com.stupidtree.hita.ui.main.timetable.views

import android.content.Context
import android.view.View
import android.view.ViewGroup
import android.widget.FrameLayout
import com.stupidtree.hita.R
import com.stupidtree.hita.data.model.timetable.TimePeriodInDay
import com.stupidtree.hita.ui.base.BaseActivity

class TimeTableBlockAddView(context: Context, var timePeriod: TimePeriodInDay) : FrameLayout(context) {


    var card: View
    var add: View
    val duration: Int
        get() = timePeriod.getLengthInMinutes()


    init {
        inflate(context, R.layout.dynamic_timetable_block_add, this)
        add = findViewById(R.id.add)
        card = findViewById(R.id.card)
        add.setOnClickListener {
            val parent = parent as ViewGroup
            parent.removeView(this@TimeTableBlockAddView)
        }
        card.setOnClickListener { add.callOnClick() }
    }
}